.github/workflows/pgdg-build.yml

```diff
@@ -85,9 +85,96 @@ jobs:
       run: |
         cd pg-oidc-validator-pgdg18 && sudo tar -czvf ../pg-oidc-validator-pgdg18.tar.gz .
 
+    - name: Upload tgz artifact
+      uses: actions/upload-artifact@v4
+      with:
+        name: pg-oidc-validator-tgz
+        path: pg-oidc-validator-pgdg18.tar.gz
+
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    needs: [pgxs-build]
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Download built package
+      uses: actions/download-artifact@v4
+      with:
+        name: pg-oidc-validator-deb
+
+    - name: Stop default PostgreSQL
+      run: |
+        sudo systemctl stop postgresql || true
+        sudo systemctl disable postgresql || true
+
+    - name: Install PostgreSQL 18
+      run: |
+        sudo sh -c 'echo "deb http://apt.postgresql.org/pub/repos/apt \
+          $(lsb_release -cs)-pgdg main 18" > /etc/apt/sources.list.d/pgdg.list'
+        sudo wget --quiet -O - \
+          https://www.postgresql.org/media/keys/ACCC4CF8.asc | \
+          sudo apt-key add -
+        sudo apt update
+        sudo apt -y install postgresql-18 libpq-oauth
```

Comment:
I personally use `apt-get` over `apt` in scripts since I have gotten the impression that `apt` has a lower guarantee of stability and friendliness to scripting but that could have been someone misinforming me.